modules/eventsource/manager_run.go

```diff
@@ -63,6 +65,31 @@ loop:
 				}
 			}
 
+			if setting.Service.EnableTimetracking {
+				usersStopwatches, err := models.GetUIDsAndStopwatch()
```

Comment:
Really what we need to do here is to be only sending the stopwatches that have changed since the last tick. This would then limits the amount of data that is pulled from the db and sent over the eventstream.

one of the unexpected problems with this solution is that we're going to be reading all of the stopwatches for perhaps up to 10 000 users.

Reply:
That would be kind of a problem, as if a user has a active stopwatch and visits a new page of Gitea, the eventsource will be the one to tell the user that it has a stopwatch, it's not cached in localstorage or something and because of the interactivity with JS it currently cannot be SSR. Given it would cache via localstorage and for some reason the user stops the stopwatch on another computer and comes back to the initial computer, it will never be told that the stopwatch has stopped. 

In order to support such architecture the user somehow needs to be told when visiting the page which stopwatches are still active and as well store stopwatches into localstorage.

> one of the unexpected problems with this solution is that we're going to be reading all of the stopwatches for perhaps up to 10 000 users.

It will be more performant than doing individual queries for all active users.

Reply:
> In order to support such architecture the user somehow needs to be told when visiting the page which stopwatches are still active and as well store stopwatches into localstorage.

so the way the notification counter gets round this is that on page load always get the current count passed in - for a stopwatch counter mechanism to work we'd need to do a similar thing but the concurrency aspects are a bit more difficult.

> It will be more performant than doing individual queries for all active users.

Certainly but we do need to consider if we can page, iterate or batch this. Iterating an SQL rows cursor seems like the best way of preventing the 10K problem I was alluding to above if we can't make a counter mechanism work (and I suspect we might not be able to make that work).

Reply:
> so the way the notification counter gets round this is that on page load always get the current count passed in - for a stopwatch counter mechanism to work we'd need to do a similar thing but the concurrency aspects are a bit more difficult.

Yeah the benefit of notification, you only need to pass a single number. For stopwatches you can see how long it's been going on, the name of it, stopping it etc. To have that seamlessly work, you will need to add a _mandatory_ database operations for signed in users when they visit a new page. With this PR I'm tyring to reduce the amount of database operations 😄.

^ Yeah we would need to add some batching to the current query.